src/Tools/ExternalAccess/Debugger/GlassTestsHotReloadService.cs

```diff
@@ -0,0 +1,140 @@
+        public async Task StartSessionAsync(Solution solution, ImmutableArray<string> capabilities, CancellationToken cancellationToken)
+        {
+            var newSessionId = await _encService.StartDebuggingSessionAsync(
+                solution,
+                new DebuggerService(capabilities),
+                captureMatchingDocuments: ImmutableArray<DocumentId>.Empty,
+                captureAllMatchingDocuments: true,
+                reportDiagnostics: false,
+                cancellationToken).ConfigureAwait(false);
+
+            Contract.ThrowIfFalse(_sessionId == default, "Session already started");
+            _sessionId = newSessionId;
+        }
+
+        /// <summary>
+        /// Emits updates for all projects that differ between the given <paramref name="solution"/> snapshot and the one given to the previous successful call 
+        /// where <paramref name="commitUpdates"/> was `true` or the one passed to <see cref="StartSessionAsync(Solution, ImmutableArray{string}, CancellationToken)"/>
+        /// for the first invocation.
+        /// </summary>
+        /// <param name="solution">Solution snapshot.</param>
+        /// <param name="commitUpdates">commits changes if true, discards if false</param>
+        /// <param name="cancellationToken">Cancellation token.</param>
+        /// <returns>
+        /// Updates (one for each changed project) and Rude Edit diagnostics. Does not include syntax or semantic diagnostics.
+        /// </returns>
+        public async Task<(ImmutableArray<ManagedModuleUpdate> updates, ImmutableArray<Diagnostic> diagnostics)> EmitSolutionUpdateAsync(Solution solution, bool commitUpdates, CancellationToken cancellationToken)
+        {
+            var sessionId = _sessionId;
+            Contract.ThrowIfFalse(sessionId != default, "Session has not started");
+
+            var results = await _encService
+                .EmitSolutionUpdateAsync(sessionId, solution, s_solutionActiveStatementSpanProvider, cancellationToken)
+                .ConfigureAwait(false);
+
+            if (results.ModuleUpdates.Status == Contracts.ManagedModuleUpdateStatus.Ready)
+            {
+                if (commitUpdates)
+                {
+                    _encService.CommitSolutionUpdate(sessionId, out _);
+                }
+                else
+                {
+                    _encService.DiscardSolutionUpdate(sessionId);
+                }
+            }
+
+            if (results.SyntaxError is not null)
+            {
+                // We do not need to acquire any updates or other
+                // diagnostics if there is a syntax error.
+                return (EmptyUpdate, EmptyDiagnostic.Add(results.SyntaxError));
+            }
+
+            var updates = results.ModuleUpdates.FromContract().Updates;
+
+            var diagnostics = await results.GetAllDiagnosticsAsync(solution, cancellationToken).ConfigureAwait(false);
+
+            return (updates, diagnostics);
```

Comment:
This is Unit Testing specific logic. We should expose something closer to EditAndContinueLanguageService

Reply:
Updated, PTAL. Wasn't sure if the debugger service should be mocked out like before, if the debugger will supply it. Let me know.

Reply:
I think this would do.